features/FEATURE_BLE/ble/BLETypes.h

```diff
@@ -528,6 +528,7 @@ struct att_security_requirement_t : SafeEnum<att_security_requirement_t, uint8_t
          */
         NONE,
 
+#if BLE_FEATURE_SECURITY
```

Comment:
I'd rather not ifdef enums - it will not help with debugging (enum values will differ based on compilation options), and has no code-size benefits

Reply:
These are to stop invalid/unhandled values. Otherwise the user might use an enum value which the code will not know what to do or more likely do something wrong. The correct solution is to grow the code by adding parameter checking for unhandled values which I didn't opt for in this instance due to time.

Reply:
also irrelevant in the light of the discussion this morning about limiting the PR to user api and init code